### DISCUSSION TITLE
Help/Advice: Exporting a table/query as a csv file from command line

### DISCUSSION THREAD

**Discussion by gaiking-uk** [sqlitebrowser/sqlitebrowser#3499] [Q&A] [answered]
## Summary
- I am trying to write a script that will export a `.db` file as a CSV

## Additional Info
- I have used the `sqlitebrowser` for a few years and this is my go-to tool for `.sqlite` and `.db` files, although I have mainly just used the GUI so far
- I read in the [SQLite Documentation](https://github.com/sqlitebrowser/sqlitebrowser/wiki/Command-Line-Interface) that I can use `-s` run a `SQL` script after opening the database
- However, while I can use this to create the table/query I want to export, I don't ***think*** it's possible to export a file (e.g. a CSV) with just `SQL`, right?

***

❌ **IF IT IS** - Can someone let me know how to do that in SQL (i.e. what the instruction is) please

✅ **IF IT ISN'T** - Is there any other way of doing this?... Ideally using `sqlitebrowser` if possible, however I also appreciate that I am maybe pushing the boundaries of what it was designed to do, so if anyone has an alternative app / approach for doing this that might be better, am happy to explore other options too.

Cheers!

**Comment by chrisjlocke** [comment]
DB4S is mainly a GUI tool for database creation, manipulation and data entry.  Exporting to CSV via command-line is a job for a command-line tool.  Nearly every program can talk to SQLite databases, so knocking up a script to do this would be quite easy .. probably easier than bending DB4S to do it.
Whats your OS of choice?

**Reply by gaiking-uk** [reply]
Hey @chrisjlocke -- Sure, that makes sense (like I say, I have been using as a GUI tool for years and it's been great, and realise it has some command line args but it's not a heavy CLI tool or anything 👍🏼)

O/S wise I am on Windows 11 (don't judge me, lol)... Any recommendations for SQLite CLI that would be good for me to check out?

**Comment by gaiking-uk** [comment, marked as answer]
## SQLite 3 (via ODBC Connector)

It turns out that I actually had an app that could do this all along...
- I use [ch-werner's SQLite ODBC driver](http://www.ch-werner.de/sqliteodbc) to enable Excel to import and query SQLite databases
- This app comes with `SQLite3`
- You can use `SQLite3` on the command line to run a query and save the output as a CSV file, for example...

```
SQLite3 -headers -csv Database.DB "SELECT * FROM table;" > Results.csv
```

**Comment by Ferretaa** [comment]
You can use the sqlite client that comes with sqlite proper. 
See Command Line Shell For SQLite: https://www.sqlite.org/cli.html
